receiver/oktareceiver/logs_test.go

```diff
@@ -0,0 +1,272 @@
+	require.NoError(t, err)
+}
+
+func TestPollPublishedAfterPollTime(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	mockDomain := "observiq.okta.com"
+	cfg.Domain = mockDomain
+
+	sink := &consumertest.LogsSink{}
+	recv, err := newOktaLogsReceiver(cfg, zap.NewNop(), sink)
+	require.NoError(t, err)
+
+	recv.client = &mockHTTPClient{
+		mockDo: func(req *http.Request) (*http.Response, error) {
+			require.Contains(t, req.URL.String(), "since=")
+			require.Contains(t, req.URL.String(), "limit="+strconv.Itoa(oktaMaxLimit))
+			if strings.Contains(req.URL.String(), "after=") {
+				return mockAPIResponseOKBasic(), nil
+			}
+			return mockAPIResponseOK1000LogsAfter(), nil
+		},
+	}
+
+	err = recv.poll(context.Background())
+	require.NoError(t, err)
+
+	logs := sink.AllLogs()
+
+	require.Equal(t, 1000, logs[0].ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().Len())
+	require.Equal(t, mockNextURL, recv.nextURL)
+
+	err = recv.Shutdown(context.Background())
+	require.NoError(t, err)
+}
+
+func newReceiver(t *testing.T, cfg *Config, c consumer.Logs) *oktaLogsReceiver {
+	r, err := newOktaLogsReceiver(cfg, zap.NewNop(), c)
+	require.NoError(t, err)
+	return r
+}
+
+func jsonFileAsPlogs(filepath string) (plog.Logs, error) {
+	expectedFileBytes, err := os.ReadFile(filepath)
+	if err != nil {
+		return plog.Logs{}, nil
+	}
+	unmarshaler := &plog.JSONUnmarshaler{}
+	return unmarshaler.UnmarshalLogs(expectedFileBytes)
+}
+
+func mockAPIResponseOKBasic() *http.Response {
+	mockRes := &http.Response{}
+	mockRes.StatusCode = http.StatusOK
+	mockRes.Header = http.Header{}
+	mockRes.Header.Add("Link", mockLinkHeaderSelf)
+	mockRes.Header.Add("Link", mockLinkHeaderNext)
+	mockRes.Body = io.NopCloser(strings.NewReader(jsonFileAsString("testdata/oktaResponseBasic.json")))
+	return mockRes
+}
+
+func mockAPIResponseOK1000Logs() *http.Response {
+	mockRes := &http.Response{}
+	mockRes.StatusCode = http.StatusOK
+	mockRes.Header = http.Header{}
+	mockRes.Header.Add("Link", mockLinkHeaderSelf)
+	mockRes.Header.Add("Link", mockLinkHeaderNext)
+	mockRes.Body = io.NopCloser(strings.NewReader(jsonFileAsString("testdata/oktaResponse1000Logs.json")))
+	return mockRes
+}
+
+func mockAPIResponseOK1000LogsAfter() *http.Response {
+	mockRes := &http.Response{}
+	mockRes.StatusCode = http.StatusOK
+	mockRes.Header = http.Header{}
+	mockRes.Header.Add("Link", mockLinkHeaderSelf)
+	mockRes.Header.Add("Link", mockLinkHeaderNext)
+	mockRes.Body = io.NopCloser(strings.NewReader(jsonFileAsString("testdata/oktaResponse1000LogsAfter.json")))
+	return mockRes
+}
+
+func mockAPIResponseTooManyRequests() *http.Response {
+	return &http.Response{
+		StatusCode: http.StatusTooManyRequests,
+		Body:       io.NopCloser(strings.NewReader("{}")),
+	}
+}
+
+func jsonFileAsString(filePath string) string {
+	jsonBytes, err := os.ReadFile(filePath)
+	if err != nil {
+		log.Fatalf("Failed to read JSON file: %s", err)
+		return ""
+	}
+	return string(jsonBytes)
+}
```

Comment:
nit: You can pass in the testing.T in function like these and use the `require` package, which makes the code a little terser.